rts/rts.cabal

```diff
@@ -689,33 +690,34 @@ library
       rts-headers,
       rts-fs
 
-    -- FIXME: by virtue of being part of the rts main library, these do not get
-    -- the flags (debug, threaded, ...) as the sub libraries.  Thus we are
-    -- likely missing -DDEBUG, -DTHREADED_RTS, etc.
-    -- One solution to this would be to turn all of these into `.h` files, and
-    -- then have the `AutoApply.cmm` in `rts-c-sources-base` include them.  This
-    -- would mean they are included in the sublibraries which will in turn apply
-    -- the sublibrary specific (c)flags.
-    autogen-cmm-sources:
-      AutoApply.cmm
-      AutoApply_V16.cmm
-
-    if arch(x86_64)
-      cmm-sources:
-        Jumps_V32.cmm (-mavx2)
-        Jumps_V64.cmm (-mavx512f)
-      autogen-cmm-sources:
-        AutoApply_V32.cmm (-mavx2)
-        AutoApply_V64.cmm (-mavx512f)
-    else
-      cmm-sources:
-        Jumps_V32.cmm
-        Jumps_V64.cmm
+    if !arch(javascript)
+      -- FIXME: by virtue of being part of the rts main library, these do not get
+      -- the flags (debug, threaded, ...) as the sub libraries.  Thus we are
+      -- likely missing -DDEBUG, -DTHREADED_RTS, etc.
+      -- One solution to this would be to turn all of these into `.h` files, and
+      -- then have the `AutoApply.cmm` in `rts-c-sources-base` include them.  This
+      -- would mean they are included in the sublibraries which will in turn apply
+      -- the sublibrary specific (c)flags.
       autogen-cmm-sources:
-        AutoApply_V32.cmm
-        AutoApply_V64.cmm
+        AutoApply.cmm
+        AutoApply_V16.cmm
 
-library ghcjs
+      if arch(x86_64)
+        cmm-sources:
+          Jumps_V32.cmm (-mavx2)
+          Jumps_V64.cmm (-mavx512f)
+        autogen-cmm-sources:
+          AutoApply_V32.cmm (-mavx2)
+          AutoApply_V64.cmm (-mavx512f)
+      else
+        cmm-sources:
+          Jumps_V32.cmm
+          Jumps_V64.cmm
+        autogen-cmm-sources:
+          AutoApply_V32.cmm
+          AutoApply_V64.cmm
+
+common ghcjs
```

Comment:
This should really be called `rts-javascript`, or `rts-ghcjs` or something like that.

Reply:
To align with the other common stanzas: `rts-js-sources` being the best?